velox/exec/Exchange.cpp

```diff
@@ -34,9 +43,9 @@ bool Exchange::getSplits(ContinueFuture* future) {
         auto remoteSplit = std::dynamic_pointer_cast<RemoteConnectorSplit>(
             split.connectorSplit);
         VELOX_CHECK(remoteSplit, "Wrong type of split");
-        exchangeClient_->addRemoteTaskId(remoteSplit->taskId);
-        ++stats_.wlock()->numSplits;
+        taskIds.push_back(remoteSplit->taskId);
       } else {
+        addTaskIds(taskIds);
```

Comment:
Are we now waiting to start fetching data until we received all splits? If so, that might be problematic.

Reply:
Not all the splits but until we fetch all the received splits. So we probably shuffle remote splits in batch but not a total shuffle

Reply:
Got it. As a followup, perhaps, we can add a Task API to fetch all available splits in one go. Thanks.